analyzers/src/SonarAnalyzer.CSharp/Rules/TestMethodShouldContainAssertion.cs

```diff
@@ -93,7 +93,11 @@ public sealed class TestMethodShouldContainAssertion : SonarDiagnosticAnalyzer
 
         private static bool ContainsAssertion(MethodDeclarationSyntax methodDeclaration, SemanticModel previousSemanticModel, ISet<IMethodSymbol> visitedSymbols, int level)
         {
-            var currentSemanticModel = methodDeclaration.EnsureCorrectSemanticModel(previousSemanticModel);
+            var currentSemanticModel = methodDeclaration.EnsureCorrectSemanticModelOrDefault(previousSemanticModel);
+            if (currentSemanticModel == null)
+            {
+                return false;
```

Comment:
won't this mean we're going to raise issues when there is this problem?  If `!ContainsAssertion` , we raise.

We should be silent if there's no semantic model. Or am I missing something?

Reply:
from offline discussion:
>It's in crossmethod inspection, we return false when we dont't inspect deeper (or we can't tell). As it's not for real life scenario, it should be fine as is